pkg/providers/aws/provider_redis.go

```diff
@@ -470,6 +471,16 @@ func buildElasticacheUpdateStrategy(elasticacheConfig *elasticache.CreateReplica
 		ec.SnapshotRetentionLimit = elasticacheConfig.SnapshotRetentionLimit
 		updateFound = true
 	}
+	for _, i := range cacheClusters {
```

Comment:
Do we want to check all found cache clusters, or should we be only checking the cache clusters which belong to our expected replication group?